src/OpenLoco/src/S5/S5.cpp

```diff
@@ -273,33 +275,413 @@ namespace OpenLoco::S5
+        dst.fixFlags = src.fixFlags;
+        dst.companyRecords = exportRecords(src.companyRecords);
+        dst.var_44C = src.var_44C;
+        dst.var_450 = src.var_450;
+        dst.var_454 = src.var_454;
+        dst.var_458 = src.var_458;
+        dst.var_45C = src.var_45C;
+        dst.var_460 = src.var_460;
+        dst.var_464 = src.var_464;
+        dst.var_468 = src.var_468;
+        dst.lastMapWindowFlags = enumValue(src.lastMapWindowAttributes.flags);
+        dst.lastMapWindowSize[0] = src.lastMapWindowAttributes.size.width;
+        dst.lastMapWindowSize[1] = src.lastMapWindowAttributes.size.height;
+        dst.lastMapWindowVar88A = src.lastMapWindowAttributes.var88A;
+        dst.lastMapWindowVar88C = src.lastMapWindowAttributes.var88C;
+
+        dst.var_478 = src.var_478;
+        dst.numMessages = src.numMessages;
+        dst.activeMessageIndex = enumValue(src.activeMessageIndex);
+        for (auto i = 0U; i < std::size(src.messages); i++)
+        {
+            dst.messages[i] = exportMessage(src.messages[i]);
+        }
+        dst.var_B94C = src.var_B94C;
+        dst.var_B950 = src.var_B950;
+        dst.var_B952 = src.var_B952;
+        dst.var_B954 = src.var_B954;
+        dst.var_B956 = src.var_B956;
+        dst.currentRainLevel = src.currentRainLevel;
+        return dst;
+    }
+
+    static S5::GameState exportGameState(const OpenLoco::GameState& src)
+    {
+        S5::GameState dst{};
+        dst.general = exportGeneralState(src);
+        for (auto i = 0U; i < std::size(src.companies); i++)
+        {
+            dst.companies[i] = exportCompany(src.companies[i]);
+        }
+        for (auto i = 0U; i < std::size(src.towns); i++)
+        {
+            dst.towns[i] = exportTown(src.towns[i]);
+        }
+        for (auto i = 0U; i < std::size(src.industries); i++)
+        {
+            dst.industries[i] = exportIndustry(src.industries[i]);
+        }
+        for (auto i = 0U; i < std::size(src.stations); i++)
+        {
+            dst.stations[i] = exportStation(src.stations[i]);
+        }
+        for (auto i = 0U; i < std::size(src.entities); i++)
+        {
+            dst.entities[i] = exportEntity(src.entities[i]);
+        }
+        for (auto i = 0U; i < std::size(src.animations); i++)
+        {
+            dst.animations[i] = exportAnimation(src.animations[i]);
+        }
+        for (auto i = 0U; i < std::size(src.waves); i++)
+        {
+            dst.waves[i] = exportWave(src.waves[i]);
+        }
+        for (auto i = 0U; i < Limits::kMaxUserStrings; i++)
+        {
+            std::ranges::copy(src.userStrings[i], dst.userStrings[i]);
+        }
+        for (auto i = 0U; i < Limits::kMaxVehicles; i++)
+        {
+            std::ranges::copy(src.routings[i], dst.routings[i]);
+        }
+        std::ranges::copy(src.orders, dst.orders);
+        return dst;
+    }
+
+    static std::unique_ptr<S5::GameState> importGameStateType2(const S5::GameStateType2& src)
```

Comment:
did you copy the design from the editor? this is exactly how i'm reading the structs there :) but since DAT format will never change, what is the reason to copy each field rather than memcpy the struct?